crates/factor-outbound-http/src/wasi.rs

```diff
@@ -204,7 +208,21 @@ async fn send_request_handler(
         return Err(ErrorCode::HttpRequestUriInvalid);
     };
 
-    let tcp_stream = timeout(connect_timeout, TcpStream::connect(&authority_str))
+    // Resolve the authority to IP addresses
+    let mut socket_addrs = tokio::net::lookup_host(&authority_str)
+        .await
+        .map_err(|_| dns_error("address not available".to_string(), 0))?
+        .collect::<Vec<_>>();
+
+    // Potentially filter out private IPs
+    if !allow_private_ips && !socket_addrs.is_empty() {
+        socket_addrs.retain(|addr| !is_private_ip(addr.ip()));
+        if socket_addrs.is_empty() {
+            return Err(ErrorCode::DestinationIpProhibited);
+        }
+    }
+
+    let tcp_stream = timeout(connect_timeout, TcpStream::connect(socket_addrs.as_slice()))
         .await
         .map_err(|_| ErrorCode::ConnectionTimeout)?
         .map_err(|err| match err.kind() {
```

Comment:
Looks like we could remove the `.starts_with("failed to lookup address information")` silliness below now.